### DISCUSSION TITLE
Having difficulty with the pricing structure

### DISCUSSION THREAD

**Discussion by j0hnm4r5** [vercel/vercel#5924] [General] [answered]
I _really_ love Next.js and Vercel (seriously, I do), but the pricing model is frankly ridiculous, and is probably going to force me to jump ship to another platform.

I run the development team for a small design agency, and we use a GitHub organization for all of our code. On staff, we have two full-time devs, and use a handful of contractors as needed. As far as I can tell, in order for Vercel to deploy any commit from a project that belongs to that org, it has to originate from the email address of a user on the Vercel team (otherwise nothing happens at all when you push; there's not even easily-visible error messages). That means for each person that _commits_ to the repo, we need to to pay $20/month. It's completely unsustainable in the long run.

And there are even workarounds to the process! We could:

- manually deploy via the CLI
- automatically deploy via the CLI with GitHub actions
- set up a webhook that triggers a new blank commit from the team owner on push and then pushes it again
- have one designated person manually commit for every deployment
- abandon GitHub Orgs altogether, move everything over to a single owner, and give access to individual repos to individual GitHub users

In essence Vercel is charging $20/committer for the luxury of clean commit histories on our own repos and/or automatic deployments on push.

I'm _extremely_ happy to pay well for a service that I'm using, but the presence of a GitHub organization is not a good determinant for the ability of people/groups to pay, and paying per repo  _committer_ is ludicrous.

**Comment by mylesthedev** [comment]
Agreed. I think Vercel seem to think Agencies hand-off the Vercel account to clients when the project is complete but in the real world you need to support it and most agencies will have multiple developers working on many sites.

I think there urgently needs to be some sort of Agency plan - even if it was more expensive!

**Comment by swarnava** [comment, marked as answer]
Hi, just wanted to let you know that this is being discussed internally. Unfortunately, I don't have an ETA on this. We are hearing your feedback and keep them coming! :)

**Reply by fabiowallner** [reply]
Any update on this?

**Reply by mylesthedev** [reply]
^

**Reply by swarnava** [reply]
Hi there. We don't have an update for this yet. Request you to keep an eye on vercel.com/changelog

**Reply by carlonoelle** [reply]
Its simply ridiculous that there is no update. Running into the same issue, really love vercel, and your analytics, but to even have viewers (the customer) to see the analytics i have to pay 20$ just for them to see the numbers, even if they dont deploy anything. Can we please get an update on this asap, otherwise i am too have to think about switching platforms

**Reply by tech-yoda** [reply]
Please update

**Comment by jonlow** [comment]
+1 on this. Current pricing structure makes it really hard for agencies to onboard clients to Vercel if you have to keep adding extra team memebrs.

**Comment by danarchos** [comment]
Wow, just been hit by this too. Paying $20 for someone to commit to a project is unrealistic. Can't agree more with the sentiment of @j0hnm4r5  - happy to pay well for services, but when you know you are being ripped off - not happy. 

Been having "internal conversations" for two years? I guess the money they are making from this mean those conversations don't go far..

EDIT: After hours of trying to make things work with Vercel - I am now using Heroku and can't believe how simple things are. It may not have the sleek designs but it just works without breaking the bank.

**Comment by NoSync** [comment]
+1. An agency could pay per project, per traffic or in any other way, but paying per committer doesn't make any sense.

**Comment by mylesthedev** [comment]
@leerob we've talked about this on X in the past but any movement/plans on this? This thread looks dead so assume it's not on anyones radar.

**Reply by leerob** [reply]
We have recently improved infra pricing, but this did not include changes around seat pricing: https://vercel.com/blog/improved-infrastructure-pricing
